### DISCUSSION TITLE
Getting suspicious counters resets

### DISCUSSION THREAD

**Discussion by Louan-Robert** [hubblo-org/scaphandre#328] [Q&A]
Hello, I am using scaphandre in one of my projects, and I am encountering some problems.

When using `scaph_socket_energy_microjoules`, `scaph_domain_energy_microjoules` or `scaph_host_energy_microjoules` metrics, I am observing suspicious counters that reset every 2 hours or, in some cases, after 15 minutes.

Note that when the counters reset, they do not drop to 0.

Here is an example using the `scaph_host_energy_microjoules` metric.
![image](https://github.com/hubblo-org/scaphandre/assets/125887532/71c5b9e4-790a-4f96-8676-5621ecf99a89)

Is this expected behavior from scaphandre?

Thank you !

**Comment by bpetit** [comment]
Hi,

This is expected behavior from RAPL counters that are reset periodically. energy_microjoules metrics are counters, they are a no-modification exposition of RAPL metrics you can find either by reading the MSRs or by reading /sys/class/powercap/*/energy_uj files

If you want to graph gauges/stable metrics in grafana/prometheus from those metrics you have to use the rate or irate operators.

If you want to get gauges directly from scaphandre metrics, use `scaph_*_power_consumption_microwatts` metrics.
